Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/ApayaTokenizationViewModel.swift

```diff
@@ -186,27 +186,32 @@ class ApayaTokenizationViewModel: PaymentMethodTokenizationViewModel, ExternalPa
         
         Primer.shared.primerRootVC?.showLoadingScreenIfNeeded(imageView: self.makeSquareLogoImageView(withDimension: 24.0), message: nil)
         
-        if PrimerDelegateProxy.isClientSessionActionsImplemented {
-            let params: [String: Any] = ["paymentMethodType": config.type.rawValue]
-            ClientSession.Action.selectPaymentMethod(resumeHandler: self, withParameters: params)
-        } else {
-            continueTokenizationFlow()
+        firstly {
+            ClientSession.Action.selectPaymentMethodWithParameters(["paymentMethodType": config.type.rawValue])
+        }
+        .done {
+            self.continueTokenizationFlow()
+        }
+        .catch { error in
+            self.handleErrorBasedOnSDKSettings(error)
         }
     }
     
     fileprivate func continueTokenizationFlow() {
+        
         do {
             try self.validate()
         } catch {
             DispatchQueue.main.async {
-                ClientSession.Action.unselectPaymentMethod(resumeHandler: nil)
-                PrimerDelegateProxy.checkoutFailed(with: error)
-                self.handleFailedTokenizationFlow(error: error)
+                self.unselectPaymentMethodWithError(error)
             }
             return
         }
         
         firstly {
+            self.handlePrimerWillCreatePaymentEvent(PaymentMethodData(type: config.type))
```

Comment:
Apaya is used only for vaulting, so this would never be performed. No problem though, it should just skip this step.

Reply:
Make a check if the delegate function is implemented.

Reply:
Addressed the delegate check ✅